sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentWriteResource.java

```diff
@@ -0,0 +1,206 @@
+
+  @Inject WriteDocumentsService documentWriteService;
+
+  @Operation(
+      description = "Create a new document. If the collection does not exist, it will be created.")
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace to write the document to."),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection to write the document to."),
+        @Parameter(
+            name = "ttl",
+            ref = OpenApiConstants.Parameters.TTL,
+            description = "The time-to-live (in seconds) of the document."),
+        @Parameter(
+            name = "profile",
+            ref = OpenApiConstants.Parameters.PROFILE,
+            description = "Include profiling information from execution."),
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "201",
+            description = "Document created. The ID will be returned.",
+            content = {
+              @Content(
+                  schema =
+                      @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                          implementation = SimpleResponseWrapper.class,
+                          properties =
+                              @SchemaProperty(name = "documentId", type = SchemaType.STRING)))
+            }),
+        @APIResponse(
+            responseCode = "404",
+            description = "Not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST)
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @POST
+  public Uni<RestResponse<Object>> createDocument(
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @QueryParam("ttl") Integer ttl,
+      @QueryParam("profile") boolean profile,
+      @NotNull JsonNode body) {
+    ExecutionContext context = ExecutionContext.create(profile);
+    return documentWriteService
+        .writeDocument(namespace, collection, body, ttl, context)
+        .onItem()
+        .transform(
+            result -> {
+              String url =
+                  String.format(
+                      "/v2/namespaces/%s/collections/%s/%s",
+                      namespace, collection, result.documentId());
```

Comment:
we should use same technique here as in `CollectionResource#createCollection`:

```
URI location = uriInfo
      .getBaseUriBuilder()
      .path(uriInfo.getPath())
      .path(collection)
      .build();
```

Reply:
* Location header should be part of the API docs
* Can you also add this to the Collection and Namespace creation please